# Pitfalls of sc/snRNA-seq


- [Heumos et al. (2023).](https://doi.org/10.1038/s41576-023-00586-w) Best practices for single-cell analysis across modalities. *Nature Reviews Genetics*, 24(8), 550–572.
- [Luecken & Theis (2019).](https://doi.org/10.15252/msb.20188746) Current best practices in single-cell RNA-seq analysis: a tutorial. *Molecular systems biology*, 15(6), e8746.
- [Smith et al. (2025).](https://doi.org/10.1038/s41586-025-09247-w) Lack of evidence for the transitional cerebellar progenitor. *Nature*, 643(8071), E1–E8.
- [Tran et al. (2020).](https://doi.org/10.1186/s13059-019-1850-9) A benchmark of batch-effect correction methods for single-cell RNA sequencing data. *Genome Biology*, 21(1), 12.


### Questions?
If you have any follow up questions about the material presented in this course, please feel free to create an [issue](https://github.com/stjude-dnb-binfcore/trainings/issues).


---

*These materials have been developed by the [DNB Bioinformatics core team](https://www.stjude.org/research/departments/developmental-neurobiology/shared-resources/bioinformatic-core.html) at the [St. Jude Children's Research Hospital](https://www.stjude.org/). These are open access materials distributed under the terms of the [BSD 2-Clause License](https://opensource.org/license/bsd-2-clause), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*