<p align="center">
  <img src="img/DNB-BINF-Core-logo.png" alt="DNB-BINF-Core logo" width="560px" />
</p>



# Intro to RNA-Seq (CAB Results) Workshop

The `Intro to RNA-Seq (CAB Results)` is an introduction to bulk RNA-Sequencing data. The course discusses applications and key steps for sequencing: library prep, sequencing technology, and data analysis. The focal point of the course are the CAB pre-processing and alignment pipeline along with how to read the reports generated by the CAB pipeline and expected QC metrics.
 
Please see the [CAB’s AutoMapper pipeline for generating alignment and expression quantification results (see Section E)](https://wiki.stjude.org/display/CAB/AutoMapper#AutoMapper-E.RNA-sequencing(RNA-seq)) for more information.

 
Moreover, you can explore the [R Shiny apps](http://ibex.stjude.org/) CAB makes available for RNA-Seq analysis.

Note: to access this page, you will need to click the link to login on the homepage and then use your St. Jude username and password to login.


## Access to the workshop recording

St. Jude employees can have access to the [recording from the workshop](https://sjcrh-my.sharepoint.com/:v:/g/personal/sfreshou_stjude_org/EfyT36KyuXdCoDEgCEz26Z8BMC7TidiHUK7-mXKtN3S0_A).

 
## Below is the main directory structure listing the analyses and data files used in this repository

```
├── gitignore.txt
├── lectures
|   ├── StJude-RNA-Sequencing-and-QC-metrics-2024-09-19.pdf
|   └── StJude-RNA-Sequencing-and-QC-metrics-2024-09-19.pptx
└── README.md
```

## Course Authors

Antonia Chroni, PhD ([@AntoniaChroni](https://github.com/AntoniaChroni))

## Contact

Contributions, issues, and feature requests are welcome! Please feel free to check [issues](https://github.com/stjudeDNBBinfCore/Trainings/issues).

---

*These materials have been developed by the [DNB Bioinformatics core team](https://www.stjude.org/research/departments/developmental-neurobiology/shared-resources/bioinformatic-core.html) at the [St. Jude Children's Research Hospital](https://www.stjude.org/). These are open access materials distributed under the terms of the [BSD 2-Clause License](https://opensource.org/license/bsd-2-clause), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*